rules.py

```diff
@@ -941,43 +941,41 @@ def check(self, element):
                 element.sourceline, element.get("objectId")))
 
 class AllCaps(base.BaseRule):
-    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
-
-    If the name elements in Candidates, Contests and Person elements are in uppercase, 
-    the list of objectIds of those elements will be returned to the user as a warning."""
+    """The Name elements in Candidates, Contests and Person elements should not
+    be in all uppercase.
 
+    If the name elements in Candidates, Contests and Person elements are in
+    uppercase, the list of objectIds of those elements will be returned to
+    the user as a warning.
+    """
+    
     def elements(self):
         return ["Candidate", "CandidateContest", "Person"]
-
+    
     def check(self, element):
         object_id = element.get("objectId")
         if element.tag == "Candidate":
-            ballot_name = element.find("BallotName")
+            ballot_name = element.find("BallotName").find("Text")
             if ballot_name is not None:
-                if ballot_name.find("Text") is not None:
-                    name = ballot_name.find("Text").text
-                    if name is not None and name == name.upper():
-                        raise base.ElectionWarning(
-                            "Line %d. Candidate %s has name in all upper case letters." % (
-                            element.sourceline, object_id))
-        elif element.tag == "Contest":
-            name_element = element.find("Name")
-            if name_element is not None:
-                name = name_element.text
-                if name is not None and name == name.upper():
+                name = ballot_name.text
+                if name and name.isupper():
                     raise base.ElectionWarning(
-                        "Line %d. Contest %s has name in all upper case letters." % (
-                        element.sourceline, object_id))
+                        "Line %d. Candidate %s has name in all upper case letters." %
+                        (element.sourceline, object_id))
+        elif element.tag == "Contest":
+            name = element.find("Name").text
+            if name and name.isupper():
+                raise base.ElectionWarning(
+                    "Line %d. Contest %s has name in all upper case letters." %
+                    (element.sourceline, object_id))
         else:
             full_name = element.find("FullName")
             if full_name is not None:
-                if full_name.find("Text") is not None:
-                    name = full_name.find("Text").text
-                    if name is not None and name == name.upper():
-                        raise base.ElectionWarning(
-                            "Line %d. Person %s has name in all upper case letters." % (
-                            element.sourceline, object_id))
-
+                name = full_name.find("Text").text
```

Comment:
FullName may not be formatted correctly to include the <Text> element, in which case the validator will run into an error.

Reply:
Updated the conditional check for this.